unity-renderer/Assets/DCLServices/EmotesCatalog/EmotesCatalogService.cs

```diff
@@ -13,10 +17,43 @@ public class EmotesCatalogService : IEmotesCatalogService
     internal readonly Dictionary<string, int> emotesOnUse = new Dictionary<string, int>();
     internal readonly Dictionary<string, HashSet<Promise<WearableItem[]>>> ownedEmotesPromisesByUser = new Dictionary<string, HashSet<Promise<WearableItem[]>>>();
 
-    public EmotesCatalogService(IEmotesCatalogBridge bridge, WearableItem[] embeddedEmotes)
+    private IAddressableResourceProvider addressableResourceProvider;
+    private EmbeddedEmotesSO embeddedEmotesSO;
+    private CancellationTokenSource addressableCTS;
+    private int retryCount = 3;
+
+    public EmotesCatalogService(IEmotesCatalogBridge bridge, IAddressableResourceProvider addressableResourceProvider)
     {
         this.bridge = bridge;
-        EmbedEmotes(embeddedEmotes);
+        this.addressableResourceProvider = addressableResourceProvider;
+        InitializeAsyncEmbeddedEmotes();
+    }
+
+    private async void InitializeAsyncEmbeddedEmotes()
+    {
+        try
+        {
+            addressableCTS = new CancellationTokenSource();
```

Comment:
I would move this cancelation token to the constructor (since it is canceled/diposed in Dispose)